mm2src/crypto/src/shared_db_id.rs

```diff
@@ -0,0 +1,35 @@
+use crate::privkey::private_from_seed_hash;
+use derive_more::Display;
+use keys::{Error as KeysError, KeyPair};
+use mm2_err_handle::prelude::*;
+use primitives::hash::H160;
+
+/// This magic string is used to change the input mnemonic passphrase the way
+/// so `sha256(mnemonic + SHARED_DB_MAGIC_SALT)`
+const SHARED_DB_MAGIC_SALT: &str = "uVa*6pcnpc9ki+VBX.6_L.";
+
+pub type SharedDbId = H160;
+
+#[derive(Display)]
+pub enum SharedDbIdError {
+    #[display(fmt = "Passphrase cannot be empty")]
+    NullStringPassphrase,
+    #[display(fmt = "Internal error: {_0}")]
+    Internal(String),
+}
+
+impl From<KeysError> for SharedDbIdError {
+    fn from(e: KeysError) -> Self { SharedDbIdError::Internal(e.to_string()) }
+}
+
+pub fn shared_db_id_from_seed(passphrase: &str) -> MmResult<SharedDbId, SharedDbIdError> {
+    let stripped_passphrase = passphrase.strip_prefix("0x").unwrap_or(passphrase);
+    if stripped_passphrase.is_empty() {
+        return MmError::err(SharedDbIdError::NullStringPassphrase);
```

Comment:
I think `EmptyPassphrase` is more understandable here

Reply:
That makes sense! Renamed this and also `MmInitError:: NullStringPassphrase` to `MmInitError:: EmptyPassphrase`